### ISSUE TITLE
getVideoInfo says orientation is landscape even if it is portrait

### ISSUE THREAD

**Issue by yzargari** [jbavari/cordova-plugin-video-editor#87]
Hello,

I'm using the cordova-plugin-media-capture plugin to record a video. After I get the file, I use the cordova-plugin-video-editor plugin to reduce video resolution and then read the dimensions, but the getVideoInfo insists all my videos are lanscape even when they're not.

If I run the getVideoInfo on the recorded file (before lowering resolution), it shows the same problem - so it has nothing to do with the transcoding process.

In addition, the width and height attributes also reflect the problem (they are showing as if the video is landscape - height is width and width is height).

Is this a bug or am I missing something?
Thanks,
Joseph.

**Comment by selsamman** [comment]
Was this on Android? I have found this as well on android.  I fixed the code in getVideInfo and think there is a similar issue on transcode where the orientation is calculated as an input to transcode.  Will submit a pull request as soon as I sort out the transcode.

**Comment by yzargari** [comment]
It was indeed on android (never checked on iOS so I don't know if that happens there too). I have figured out a reasonable workaround, but am still looking forward to test your fix.

Thanks for the help!

**Comment by selsamman** [comment]
So the fix seems to work OK for me for the getVideoInfo.  I applied the fix to the transcode as well since it seems to need to determine the orientation.  However I am running into a problem with HD videos.  Transcoding a HD 1920x1080 videos fails within the transcode library whenever the orientation is set as landscape.  Because of the way the code was before it was always specifying landscape.  So it  works now on at least portrait HD video.  So I am still trying to see exactly what is going on there. Have to get back to my day job so won't get much time before Friday to work on this.  I want to get a little further on this before submitting an official pull request but you  can see if the fix works for you include the plugin as https://github.com/selsamman/cordova-plugin-video-editor#b5ea9b2ba1b887088f6c3c9939609fd7ebef7fed

**Comment by selsamman** [comment]
Made some changes and tested portrait and landscape transcoding and confirmed input and output are showing the correct orientation on getVideoInfo.  I will submit as a pull request this commit: 

https://github.com/selsamman/cordova-plugin-video-editor#6c2635562dc29f88b8beec5a8baeff84d80304cb

One thing that confused me was that you can't transcode to just any height.  I believe it must be one of the standard ones supported.

**Comment by BilluBong** [comment]
Hi,
I was facing same issue.
I changed few code at line no 238 in `**VideoEditor.java**

`if (mmrOrientation == "0" || mmrOrientation == "180") {
       orientation = "portrait";
} else {
       orientation = "landscape"; 
}`

to 

`if (mmrOrientation.equals("0") || mmrOrientation.equals("180")) { 
                        orientation = "landscape"; 
                    } else {
                        orientation = "portrait";
                     }`

**Comment by BilluBong** [comment]
`var res = gApp.uploadVideo(videos); function onprogress(percent) {$timeout(function () {res.uploadpercent = percent;});}`

**Comment by BilluBong** [comment]
Hi, I was facing same issue. I changed few code at line no 238 in **VideoEditor.java**
`if (mmrOrientation == "0" || mmrOrientation == "180") { orientation = "portrait"; } else { orientation = "landscape";  }`

to 

`if (mmrOrientation.equals("0") || mmrOrientation.equals("180")) {  orientation = "landscape";  } else { orientation = "portrait";  }`

**Comment by selsamman** [comment]
Yes that is essentially what is in the PR that was recently merged.
